src/deploy.ts

```diff
@@ -199,77 +204,79 @@ export async function deploy(
 
   let buildFilePaths: string[] | null = null;
 
-  let doBuild = false;
-  let youngestAge;
+  let doBuild = force === "build";
```

Comment:
I think we could simplify this as:

```js
  const doBuild =
    force === "build"
      ? true // always build
      : force === "deploy"
      ? false // never build, and error if build is missing
      : effects.isTty
      ? await askWhetherToBuild() // prompt the user
      : isBuildMissing(); // automatically build if missing
```

This would eliminate the arbitrary 5-minute definition of stale in favor of always prompting when possible (and otherwise only building if the build is missing), and you can bypass the prompt using `--build` or `--no-build` as appropriate. It’s more complicated if we want to be backwards-compatible with `--if-stale` and `--if-missing`, though. 😓